_posts/2016-4-2-code-splitting-angular2.md

```diff
@@ -94,59 +149,70 @@ export default class DynamicSubComp {}
 }
 ```
 
-Notice that we're also importing a `.scss` file. This is made possible by using the [sass-loader](https://github.com/jtangelder/sass-loader). The Sass imported into this module will be bundled together with the JS and loaded on demand as well.
+Using [Angular 2's modules](https://angular.io/docs/ts/latest/guide/ngmodule.html), we can encapsulate sub routes and dependencies in NgModules. In this way, when we resolve this chunk, we are also resolving its code dependencies as well as defining all of the sub routes associated with this module.
+
+Notice also that we're additionally importing a `.scss` file. This is made possible by using the [sass-loader](https://github.com/jtangelder/sass-loader). The Sass imported into this module will be bundled together with the JS and loaded on demand as well.
 
-Let's add the link in the App component for users to click to add the dynamic component.
+Let's add the link in the inline component for users to click to add the dynamic component.
 
 ```js
 @Component({
-  selector: 'yo',
+  selector: 'inline-comp',
   template: `
-    <div>
-      <h1>Code Splitting in Angular 2</h1>
-      <div>
-        <a [routerLink]="['LoadDynamicSub']">Load Sub Component</a>
-      </div>
-      <router-outlet></router-outlet>
-    </div>
-  `,
-  directives: [ ROUTER_DIRECTIVES ]
+  <div>
+    <p>This component is the default component and was loaded inline.</p>
+    <a routerLink="/dynamic" routerLinkActive="active">Load a dynamic component</a>
+  </div>`
 })
+export default class InlineComp {}
 ```
 
-By defining an [AsyncRoute](https://angular.io/docs/js/latest/api/router/AsyncRoute-class.html) in our RouteConfig decorator, we just need to provide a function that returns a promise as the `loader` config for that route. Now, we have a place to use the `System.import()` syntax to return a promise that contains our component.
+Note that the dynamic component will eventually be rendered through the <router-outlet> tag in AppComponent.
+
+By defining a route with a `loadChildren` property, we have a handle to load another Angular module asynchronously. Now, we have a place to use the `System.import()` syntax to return a promise that contains our component.
 
 ```js
-// App.ts
-new AsyncRoute({
-  path: '/sub',
-  loader: () => System.import('./DynamicSubComp').then((comp: any) => {
-    return comp.default;
-  }),
-  name: 'LoadDynamicSub'
-})
+// Routing.ts
+const appRoutes: Routes = [
+  {
+    path: 'dynamic',
+    loadChildren: () => {
+      return System.import('./DynamicSubComp').then((comp: any) => {
```

Comment:
I updated to `System.import('./dynamic.module')`

```
could you just export DynamicModule and then here do return System.import('./dynamic.module.ts') without worrying about the default?
```

making `DynamicModule` have
```
export DynamicModule
```
would require you to do `return comp.DynamicModule` since that is in effect naming that export to be the class. As mentioned in the following section in the blog post, you need to specify which `export` you want. For files exporting a single class, seems like it makes sense to me to keep it `default`. Hopefully that clears it up enough for people why you need the `default`